packages/analytics/analytics-geo-map/src/components/AnalyticsGeoMap.vue

```diff
@@ -0,0 +1,377 @@
+    const minLong = Math.min(...longs)
+    const maxLong = Math.max(...longs)
+    map.value?.fitBounds([
+      [minLong, minLat],
+      [maxLong, maxLat],
+    ])
+  }
+}
+
+const mapOptions = computed(() => {
+  const options: MapOptions = {
+    container: 'mapContainer',
+    style: { version: 8, sources: {}, layers: [] },
+  }
+  if (!props.center && !props.initialZoom) {
+    options.bounds = [
+      [-180, -90],
+      [180, 90],
+    ]
+  } else {
+    if (props.center) {
+      options.center = props.center as LngLatLike
+    }
+    if (props.initialZoom) {
+      options.zoom = props.initialZoom
+    }
+  }
+
+  return options
+})
+
+onMounted(() => {
+
+  map.value = new Map(mapOptions.value)
+  map.value.on('load', () => {
+    map.value?.addSource('countries', {
+      type: 'geojson',
+      data: props.geoJsonData,
+    })
+
+    map.value?.addLayer({
+      id: 'countries-layer',
+      type: 'fill',
+      source: 'countries',
+      paint: layerPaint.value,
+    })
+
+    const popup = new Popup({
+      closeButton: false,
+      closeOnClick: false,
+    })
+
+    map.value?.on('mousemove', 'countries-layer', (e) => {
+      const feature = e.features?.[0]
+      if (feature) {
+        const { iso_a2, admin } = feature.properties
+        const metric = props.countryMetrics[iso_a2]
+        if (metric !== undefined) {
+          // @ts-ignore - dynamic i18n key
+          const popupHtml = props.showTooltipValue ? `<strong>${admin}</strong>: ${approxNum(metric, { capital: true })} ${i18n.t(`metricUnits.${props.metricUnit}`, { plural: metric.length > 1 ? 's' : '' })}` : `<strong>${admin}</strong>`
+          popup.setLngLat(e.lngLat).setHTML(popupHtml).addTo(map.value as Map)
+        } else {
+          popup.remove()
+        }
+      }
+    })
+
+    map.value?.on('mouseleave', 'countries-layer', () => {
+      popup.remove()
+    })
+
+    if (props.fitToCountry) {
+      goToCountry(props.fitToCountry)
+    }
+
+  })
+})
+
+watch(() => props.countryMetrics, () => {
+  if (map.value && map.value.isStyleLoaded()) {
+    if (map.value.getLayer('countries-layer')) {
+      map.value.removeLayer('countries-layer')
+    }
+    map.value.addLayer({
+      id: 'countries-layer',
+      type: 'fill',
+      source: 'countries',
+      paint: layerPaint.value,
+    })
+  }
+})
+
+watch(() => props.fitToCountry, (newVal) => {
+  if (map.value && newVal) {
+    goToCountry(newVal)
+  } else {
+    if (!mapOptions.value.center && !mapOptions.value.zoom) {
+      map.value?.fitBounds([
+        [-180, -90],
+        [180, 90],
+      ])
+    } else {
+      if (mapOptions.value.zoom) {
+        map.value?.setZoom(mapOptions.value.zoom)
+      }
+      if (mapOptions.value.center) {
+        map.value?.flyTo({ center: mapOptions.value.center })
+      }
+    }
+  }
+})
+
+</script>
+
+<style lang="scss" scoped>
+.kong-ui-public-analytics-geo-map {
+  height: 100%;
+  position: relative;
+  width: 100%;
+
```

Comment:
Do you mind adding this to hide the MapLibre attribution?

```
&:deep(.maplibregl-control-container) {
    .maplibregl-ctrl-attrib {
      display: none;
    }
  }
 ```

Reply:
done